### DISCUSSION TITLE
why dependabot PR should modify package-lock.json / yarn.lock ?

### DISCUSSION THREAD

**Discussion by jeffschulzusc** [community/community#43754] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

conceptually, i don't understand that dependabot should modify package-lock.json or yarn.lock ...

ex.
https://classic.yarnpkg.com/lang/en/docs/yarn-lock/
_"Your yarn.lock file is auto-generated and **should be handled entirely by Yarn**. As you add/upgrade/remove dependencies with the Yarn CLI, it will automatically update your yarn.lock file. **Do not edit this file directly as it is easy to break something**."_

dependabot alerts with security level are quite valuable, but i question any automatically created dependabot PR that purports
to change package-lock.json / yarn.lock, and currently i have to manually examine each PR in detail and i usually manually apply package.json changes 
and close the PR ... 

given a security alert, is it not the responsibility of a package's owner that is up the dependency tree to update their package ?

is there a configuration in dependabot.yml to tell dependabot not to make changes to package-lock.json / yarn.lock ?

thanks :)

**Comment by ferrybig** [comment]
Your upstream dependencies just say that any version of the dependency is suited. Dependabot updates the lock file so a non vulnerable version of the dependency is locked and downloaded

**Comment by jeffschulzusc** [comment]
hi thanks for the input ...
 i maintain that "any version of the dependency is suited" is a problem of the package owner, they should update their package code
